src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,343 @@
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < files.size(); i++) {
+            sb.append(fileInfo(files.get(i)));
+        }
+        return sb.toString();
+
+    }
+
+    private FileFilter fileFilterForAll() {
+        FileFilter fileFilterForAll = new FileFilter() {
+            @Override
+            public boolean accept(File pathname) {
+                return true;
+            }
+        };
+        return fileFilterForAll;
+    }
+
+
+    Comparator comparatorForLexAscending = new Comparator<File>() {
+        @Override
+        public int compare(File o1, File o2) {
+            if (o1.isDirectory())
+                return o2.isDirectory() ? o1.compareTo(o2) : -1;
+            else if (o2.isDirectory())
+                return 1;
+            return o1.compareTo(o2);
+        }
+
+    };
+    Comparator comparatorForLexAscendingNoOrder = new Comparator<File>() {
+        @Override
+        public int compare(File o1, File o2) {
+            return String.CASE_INSENSITIVE_ORDER.compare(o1.getName(), o2.getName());
+        }
+    };
+    Comparator comparatorForLexDescending = new Comparator<File>() {
+        @Override
+        public int compare(File o1, File o2) {
+            if (o1.isDirectory())
+                return o2.isDirectory() ? o2.compareTo(o1) : -1;
+            else if (o2.isDirectory())
+                return 1;
+            return o2.compareTo(o1);
+        }
+    };
+    Comparator comparatorForLexDescendingNoOrder = new Comparator<File>() {
+        @Override
+        public int compare(File o1, File o2) {
+            return String.CASE_INSENSITIVE_ORDER.compare(o2.getName(), o1.getName());
+        }
+    };
+    Comparator comparatorForLastUpdateDescending = new Comparator<File>() {
+        @Override
+        public int compare(File o1, File o2) {
+            if (o1.isDirectory())
+                return o2.isDirectory() ? Long.valueOf(o1.lastModified()).compareTo(o2.lastModified()) : -1;
+            else if (o2.isDirectory())
+                return 1;
+            return Long.valueOf(o1.lastModified()).compareTo(o2.lastModified());
+        }
+    };
+    Comparator comparatorForLastUpdateDescendingNoOrder = new Comparator<File>() {
+        @Override
+        public int compare(File o1, File o2) {
+            return Long.valueOf(o1.lastModified()).compareTo(o2.lastModified());
+        }
+    };
+    Comparator comparatorForLastUpdateAscending = new Comparator<File>() {
+        @Override
+        public int compare(File o1, File o2) {
+            if (o1.isDirectory())
+                return o2.isDirectory() ? Long.valueOf(o2.lastModified()).compareTo(o1.lastModified()) : -1;
+            else if (o2.isDirectory())
+                return 1;
+            return Long.valueOf(o2.lastModified()).compareTo(o1.lastModified());
+        }
+    };
+    Comparator comparatorForLastUpdateAscendingNoOrder = new Comparator<File>() {
```

Comment:
such fields(all your comparators) should `private final` cause there is no need to use them elsewhere.